core/sys/types/build.rs

```diff
@@ -24,11 +24,15 @@ fn sgx_library_path() -> String {
 
 fn main() {
     let bindings = Builder::default()
-        .header_contents("core_types.h", "#include <sgx_error.h>")
+        .header_contents(
+            "core_types.h",
+            "#include <sgx_error.h>\n#include <sgx_report.h>",
+        )
```

Comment:
Are you suggesting removing the `mc-sgx-urts-sys-types` crate?
```rust
        .allowlist_type("sgx_enclave_id_t")
        .allowlist_type("sgx_launch_token_t")
        .allowlist_type("sgx_misc_attribute_t")
```
I'm fine with that if that's the desire.  I thought the `core` types were meant to house those types common to both trusted and untrusted code.

Reply:
For example: functions that live in `libsgx_dcap_ql.so` would get their bindgen prototypes in `mc-sgx-dcap-ql-sys`, and functions that live in `libsgx_epid.so` would get their bindgen prototypes in `mc-sgx-epid-sys`. But if they both use, e.g. `sgx_quote_t`, or `sgx_report_t`, then those would both go into `mc-sgx-core-sys-types`, which `-epid-sys` and `-dcap-ql-sys` both depend on.

That said, what I'm suggesting here is that there's a single `.h` file that this library uses, and that file is the one doing the `#include <sgx_foo.h>`s, so we don't need to edit the build script repeatedly.

Reply:
> what I'm suggesting here is that there's a single .h file that this library uses, and that file is the one doing the #include <sgx_foo.h>s, so we don't need to edit the build script repeatedly.

So you're saying instead of using [header_contents()](https://docs.rs/bindgen/latest/bindgen/struct.Builder.html#method.header_contents) we create a real header file, which we update as needed, and instead use, [header()](https://docs.rs/bindgen/latest/bindgen/struct.Builder.html#method.header)